src/main/java/com/celements/store/CelHibernateStore.java

```diff
@@ -0,0 +1,101 @@
+package com.celements.store;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.ImmutableDocumentReference;
+
+import com.google.common.collect.Iterables;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.store.XWikiHibernateStore;
+
+@Singleton
+@Component
+public class CelHibernateStore extends XWikiHibernateStore {
+
+  @Override
+  public void saveXWikiDoc(XWikiDocument doc, final XWikiContext context,
+      final boolean bTransaction) throws XWikiException {
+    // XWikiHibernateStore.saveXWikiDoc requires a mutable docRef
+    DocRefMutabilityExecutor<Void> exec = new DocRefMutabilityExecutor<Void>() {
+
+      @Override
+      public Void call(XWikiDocument doc) throws XWikiException {
+        CelHibernateStore.super.saveXWikiDoc(doc, context, bTransaction);

```

Comment:
Why not just copy the saveXWikiDoc implementation and adjust the setWikiReference line with a doc.setDocumentReference(References.adjustRef(... ?